backend/src/main/java/com/redhat/cloud/notifications/routers/sources/SourcesService.java

```diff
@@ -0,0 +1,84 @@
+package com.redhat.cloud.notifications.routers.sources;
+
+import io.quarkus.rest.client.reactive.ClientExceptionMapper;
+import org.eclipse.microprofile.rest.client.annotation.RegisterClientHeaders;
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PATCH;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.Response;
+
+/**
+ * <p>REST Client for the Sources API. The OpenAPI spec is available at:</p>
+ *
+ * <ul>
+ *  <li><a href="https://console.stage.redhat.com/api/sources/v3.1/openapi.json">OpenApi v3.1 in the stage environment.</a></li>
+ *  <li><a href="https://console.redhat.com/api/sources/v3.1/openapi.json">OpenApi v3.1 in the production environment.</a></li>
+ *  <li><a href="https://github.com/RedHatInsights/sources-api-go/blob/main/public/openapi-3-v3.1.json">OpenApi v3.1 on GitHub.</a></li>
+ * </ul>
+ *
+ * <p>Please be aware of the following:</p>
+ *
+ * <ul>
+ *     <li>If sources is using a database backend, only the {@link Secret#password} field will get encrypted.</li>
+ *     <li>On the other hand, if sources is using the AWS Secrets Manager, then the whole secret will get encrypted.</li>
+ * </ul>
+ */
+@ApplicationScoped
+@RegisterClientHeaders(SourcesXRHIDHeaderFactory.class)
+@RegisterRestClient(configKey = "sources")
+public interface SourcesService {
```

Comment:
I added the fault tolerance bit so that we can be sure that failures are not due to "one-off" errors. I added the extension by running the `quarkus extension add 'smallrye-fault-tolerance'` command. That seemed to stir the `pom.xml` file a bit. Is this okay or do you usually add extension manually?

Reply:
Done as well! I've created a ticket for the CLI stuff: [RHCLOUD-22602](https://issues.redhat.com/browse/RHCLOUD-22602)